torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4028,6 +4058,48 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                    dtypesIfCUDA=all_types_and(torch.bool, torch.half, torch.bfloat16),
                    sample_inputs_func=sample_inputs_logit,
                    safe_casts_outputs=True),
+    OpInfo('dot',
+           dtypes=floating_and_complex_types_and(torch.float16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16),
+           skips=(
+               # dot does not handle correctly out= dtypes
+               # https://github.com/pytorch/pytorch/issues/55561
+               SkipInfo('TestCommon', 'test_out'),
+           ),
+           assert_autodiffed=True,
+           sample_inputs_func=sample_inputs_dot_vdot),
+    OpInfo('vdot',
+           dtypes=all_types_and_complex_and(torch.float16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16),
+           skips=(
+               # vdot does not handle correctly out= dtypes
+               # https://github.com/pytorch/pytorch/issues/55561
+               SkipInfo('TestCommon', 'test_out'),
+           ),
+           sample_inputs_func=sample_inputs_dot_vdot),
+    OpInfo('bmm',
+           dtypes=all_types_and_complex_and(torch.bfloat16, torch.float16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.bfloat16, torch.float16),
+           assert_autodiffed=True,
+           skips=(
+               # bmm does not correctly warn when resizing out= inputs
+               SkipInfo('TestCommon', 'test_out'),
+               # cuda gradchecks are slow
+               # see discussion https://github.com/pytorch/pytorch/pull/47761#issuecomment-747316775
+               SkipInfo('TestGradients', 'test_fn_gradgrad', device_type='cuda'),),
+           sample_inputs_func=sample_inputs_bmm),
+    OpInfo('mv',
+           dtypes=all_types_and_complex_and(torch.float16, torch.bfloat16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.bfloat16, torch.float16),
+           skips=(
+               # bmm does not correctly warn when resizing out= inputs
+               SkipInfo('TestCommon', 'test_out'),
+               SkipInfo('TestOpInfo', 'test_supported_backward', dtypes=(torch.float16,)),
```

Comment:
Suggestion: instead of skipping these two tests, `dtypes` should be `all_types_and_complex_and(torch.bfloat16)`. But if these tests also runs on ROCM, then maybe `dtypesIfROCM` would've to be specified as well.

Reply:
@imaginary-person I think the issue here is just that the backward is not supported for `torch.float16`, but the forward is, so skipping the backward seemed reasonable?

Reply:
@anjali411, but you're also skipping `test_supported_dtypes` for `torch.float16`.

Reply:
> @anjali411, but you're also skipping `test_supported_dtypes` for `torch.float16`.

that's because `mv` which calls into addmv which doesn't fully support float16 but does support it in some cases.

Reply:
@anjali411, oh, I would've missed this nuance.

Reply:
@anjali411, it looks like `addmv` hasn't been enabled for `float16` at all on [CPU](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Blas.cpp#L22), but is supported on [CUDA](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Blas.cu#L15).
So maybe these 2 tests should only be skipped on CPU?

Reply:
@anjali411, ah I see now what you meant. Something like #55755.